packages/did-provider-jwk/src/jwkDidUtils.ts

```diff
@@ -0,0 +1,53 @@
+import type { JwkDidSupportedKeyTypes } from '.'
+import { VerificationMethod, type JsonWebKey } from 'did-resolver'
+import { hexToBytes, bytesToBase64url, extractPublicKeyHex } from '@veramo/utils'
+import elliptic from 'elliptic'
+
+function createJWK(keyType: string, pubKey: string | Uint8Array): JsonWebKey | undefined {
+  if(keyType === 'Secp256k1') {
+    const EC = new elliptic.ec('secp256k1')
+    const pubPoint = EC.keyFromPublic(pubKey, 'hex').getPublic()
+    const x = pubPoint.getX()
+    const y = pubPoint.getY()
+
+    return {
+      crv: 'secp256k1',
+      kty: 'EC',
+      x: bytesToBase64url(hexToBytes(x.toString('hex'))),
+      y: bytesToBase64url(hexToBytes(y.toString('hex'))),
+    } as JsonWebKey
+  }
+  if(keyType === 'Secp256r1') {
+    const EC = new elliptic.ec('p256')
+    // add '03' prefix to public key
```

Comment:
Where is this prefix coming from?
Can you please share a link to some reference?

Reply:
We do it in our implementation as well (compressed format):
https://github.com/Sphereon-Opensource/ssi-sdk/blob/develop/packages/jwk-did-provider/src/functions.ts#L113

Reply:
It is added to indicate a compressed public key, otherwise we have an invalid hex string. The prefix (02 if the `y` coordinate of the public key point is even, 03 if the `y` is odd) is removed in the Veramo kms: https://github.com/uport-project/veramo/blob/abfba31ff8497f00cddec5ca7a561fe32b1ab0bb/packages/kms-local/src/key-management-system.ts#L333

Reply:
hmm.. this seems odd (no pun intended :) )
Veramo kms should not remove this prefix. At most, it should remove the `0x` prefix from hex encoded public keys, but not the parity.
If it's removing the parity it is a bug.

Reply:
Oh, it really is removing the parity (https://github.com/uport-project/veramo/blob/9c73d98fd217ed9a612767f49a235cdbf43619cb/packages/kms-local/src/key-management-system.ts#L333).

@nklomp I missed this in my original review. Can you share some insight for why this would be needed?

Reply:
It shouldn't. I think we might need to look at how to fix it for any keys already stored though

Reply:
I can add a new script to the `@veramo/data-store#migrations`, but it will only cover the TypeORM based storage.

Reply:
oh.. actually, it will have to be done manually because the TypeORM migrations don't have access to the decryption key to get the private keys and recompute the correct public key :(

Reply:
Yeah, I think we can do it in code as well. A simple check for the prefix on a Secp256r1 key should suffice

Reply:
I created a PR with the fix and proposed a method to recompute the keys: #1137

I'd appreciate your review @nklomp